kits/arm/ex_AR_kit.cpp

```diff
@@ -0,0 +1,223 @@
+#include <thread>
+#include "hebi_util.hpp"
+
+using namespace hebi;
+using namespace experimental; // For all things mobileIO 
+
+
+
+Eigen::Matrix3d makeRotationMatrix (hebi::Quaternionf phone_orientation) {
+  Eigen::Quaterniond q;
+  q.w() = phone_orientation.getW();
+  q.x() = phone_orientation.getX();
+  q.y() = phone_orientation.getY();
+  q.z() = phone_orientation.getZ();
+  return q.toRotationMatrix();
+}
+
+
+int main(int argc, char* argv[])
+{
+  //////////////////////////
+  ///// Config Setup ///////
+  //////////////////////////
+
+  // Config file path
+  const std::string example_config_file = "config/ex_AR_kit.cfg.yaml";
+  std::vector<std::string> errors;
+  
+  // Load the config
+  const auto example_config = RobotConfig::loadConfig(example_config_file, errors);
+  for (const auto& error : errors) {
+    std::cerr << error << std::endl;
+  }
+  if (!example_config) {
+    std::cerr << "Failed to load configuration from: " << example_config_file << std::endl;
+    return -1;
+  }
+
+  // For this demo, we need the arm and mobile_io
+  std::unique_ptr<hebi::experimental::arm::Arm> arm;
+  std::unique_ptr<hebi::util::MobileIO> mobile_io;
+
+  //////////////////////////
+  ///// Arm Setup //////////
+  //////////////////////////
+
+  // Create the arm object from the configuration
+  arm = hebi::experimental::arm::Arm::create(*example_config);
+
+  // Keep retrying if arm not found
+  while (!arm) {
+      std::cerr << "Failed to create arm, retrying..." << std::endl;
+
+      // Wait for 1 second before retrying
+      std::this_thread::sleep_for(std::chrono::seconds(1));  
+
+      // Retry
+      arm = hebi::experimental::arm::Arm::create(*example_config);
+  }
+  std::cout << "Arm connected." << std::endl;
+
+  //////////////////////////
+  //// MobileIO Setup //////
+  //////////////////////////
+
+  // Create the mobile_io object from the configuration
+  std::cout << "Waiting for Mobile IO device to come online..." << std::endl;
+  mobile_io = createMobileIOFromConfig(*example_config, example_config_file);
+
+  // Keep retrying if Mobile IO not found
+  while (mobile_io == nullptr) {
+      std::cout << "Couldn't find Mobile IO. Check name, family, or device status..." << std::endl;
+
+      // Wait for 1 second before retrying
+      std::this_thread::sleep_for(std::chrono::seconds(1));  
+
+      // Retry
+      mobile_io = createMobileIOFromConfig(*example_config, example_config_file);
+  }
+  std::cout << "Mobile IO connected." << std::endl;
+
+  // Setup instructions for display
+  std::string instructions;
+  instructions = ("🏠 - Home Position\n📲 - AR Control Mode\n"
+                  "🌍 - Grav Comp Mode\n❌ - Quit\n");
+
+  // Clear any garbage on screen
+  mobile_io->clearText();
+
+  // Display instructions on screen
+  mobile_io->appendText(instructions); 
+
+  // Setup state variable for mobile device
+  auto last_mobile_state = mobile_io->update();
+
+  //////////////////////////
+  //// Main Control Loop ///
```

Comment:
Huh -- before the changes, we were at line 86 at this point.  Code length isn't the most important metric, but if done well (and still appropriately commented) shorter code can improve readability.  I had expected the examples to get a bit shorter by using the layout and config files, so I was a little surprised.

I think a bit of this is just slightly different style, and some is a printing out more error info when loading configs, but I'm curious whether you see anything I'm missing, or if you have ideas to improve this.  I'd love to eventually get these 110 lines down to something simple and relatively self-documenting like the following, but in the meantime I wonder if we could maintain the readability, but shorten some of the boilerplate?

```
auto config = Config::create('file', ErrorMode=PrintAll);
if (!config)
  return 1;
auto arm = Arm::createWithRetries(config);
auto mobile_io = MobileIO::createWithRetries(config);
if (!arm || !mobile_io)
  return 1;
```